### PR TITLE
[93] Updated response formatting interceptor 

### LINKED ISSUE #141: update response formatting interceptor

### PR DESCRIPTION
**Pull request by Savio629**
## Updated response formatting interceptor 

fixes https://github.com/SamagraX-Stencil/stencil/issues/141

### Demo

image

image

## Summary by CodeRabbit

- **New Features**
  - Enhanced response formatting to include `success` flag, `message`, `data`, and `error` fields for better clarity.

- **Bug Fixes**
  - Improved error handling and response formatting logic to provide more informative error messages.

- **Tests**
  - Added comprehensive tests for file upload handling to ensure robust functionality and error management.

### CODE REVIEW

File: packages/common/src/interceptors/response-format.interceptor.ts [hunk 1]
```diff
@@ -14,46 +14,36 @@ export class ResponseFormatInterceptor implements NestInterceptor {
...
-            data: {
-              message: data,
-              statusCode: context.switchToHttp().getResponse().statusCode,
-            },
-            errors: [],
-          },
-        ],
+        success: true,
+        statusCode: context.switchToHttp().getResponse().statusCode,
+        message: context.switchToHttp().getRequest().method === 'POST' ? 'Created' : 'Success',
```
Comment (techsavvyash) [inline packages/common/src/interceptors/response-format.interceptor.ts:19]: this is to be removed.

File: packages/common/src/interceptors/response-format.interceptor.ts [hunk 2]
```diff
@@ -14,46 +14,36 @@ export class ResponseFormatInterceptor implements NestInterceptor {
...
+        data,
+        error: {},
```
Comment (techsavvyash) [inline packages/common/src/interceptors/response-format.interceptor.ts:21]: make error null

File: packages/common/src/interceptors/response-format.interceptor.ts [hunk 3]
```diff
@@ -14,46 +14,36 @@ export class ResponseFormatInterceptor implements NestInterceptor {
...
-          {
-            statuscode: status,
-            data: {},
-            errors: [this.formatError(error)],
-          },
-        ]);
+        return of({
+          success: false,
+          statusCode: status,
+          message,
```
Comment (techsavvyash) [inline packages/common/src/interceptors/response-format.interceptor.ts:30]: remove this.

File: packages/common/src/interceptors/response-format.interceptor.ts [hunk 4]
```diff
@@ -14,46 +14,36 @@ export class ResponseFormatInterceptor implements NestInterceptor {
...
+          data: null,
+          error: this.formatError(status, message),
```
Comment (techsavvyash) [inline packages/common/src/interceptors/response-format.interceptor.ts:32]: don't send status again

**Review by techsavvyash — CHANGES_REQUESTED** [review]

**Review by techsavvyash — APPROVED** [review]

**Comment by Savio629** [comment]
Parent issue: #93

**Pull request merged**